player/lua/console.lua

```diff
@@ -36,6 +36,8 @@ function detect_platform()
         return 'windows'
     elseif mp.get_property_native('options/macos-force-dedicated-gpu', o) ~= o then
         return 'macos'
+    elseif os.getenv('WAYLAND_DISPLAY') then
+        return 'wayland'
```

Comment:
The problem with this is that someone could have a wayland compositor running and x11 at the same time and it would always pick the wayland clipboard. Instead, I think you should be able to check one of the wayland options.

Reply:
I tried that initally but all the Wayland options (wayand-app-id, wayland-disable-vsync, wayland-edge-pixels-pointer, wayland-edge-pixels-touch) are interpreted and set correctly on X11, they're not like the Windows and macOS option that don't even get compiled in, so I don't know of any other way.
Wouldn't what you described happen only if you run mpv in XWayland? wl-paste should still work in that case.

Reply:
Well `wl-paste` working correctly would depend on your compositor to handle it right. But you're right, I forgot that the wayland options do get defined anyway (in most cases). This is probably the best way short of explicitly checking the vo/gpu-context, but that's way overkill.